source/org/zfin/marker/repository/HibernateMarkerRepository.java

```diff
@@ -793,6 +794,57 @@ public void addDataAliasAttribution(DataAlias alias, Publication attribution, Ma
         infrastructureRepository.insertUpdatesTable(marker, "", "new attribution, data alias: " + alias.getAlias() + " with pub: " + attributionZdbID, attributionZdbID, "");
     }
 
+    public void addGenomeLocationAttribution(Location genomeLocation, String publicationID) {
+        Publication publication = getPublicationRepository().getPublication(publicationID);
+        this.addGenomeLocationAttribution(genomeLocation, publication);
+    }
+
+    public void addGenomeLocationAttribution(Location genomeLocation, Publication attribution) {
+
+        String attributionZdbID = attribution.getZdbID();
+        String relZdbID = genomeLocation.getZdbID();
+
+        if (attributionZdbID.equals("")) {
+            throw new RuntimeException("Cannot attribute this location with a blank pub.");
+        }
+
+        RecordAttribution recordAttribution = infrastructureRepository.getRecordAttribution(relZdbID, attributionZdbID, RecordAttribution.SourceType.STANDARD);
+
+        // only add the publication when it is not there
+        if (recordAttribution == null) {
+            PublicationAttribution pa = new PublicationAttribution();
+            pa.setSourceZdbID(attributionZdbID);
+            pa.setDataZdbID(relZdbID);
+            pa.setSourceType(RecordAttribution.SourceType.STANDARD);
+            pa.setPublication(attribution);
+            currentSession().save(pa);
+            currentSession().refresh(genomeLocation);
+
+            infrastructureRepository.insertUpdatesTable(genomeLocation.getZdbID(), "", "new attribution publication to genome location " + attributionZdbID + " to " + relZdbID);
+        }
+
+    }
+
+    public void synchronizeGenomeLocationAttributions(MarkerLocation genomeLocation, Set<String> publicationIDsToSync) {
+        //ADD
+        for (String publicationID : publicationIDsToSync) {
+            this.addGenomeLocationAttribution(genomeLocation, publicationID);

```

Comment:
do you need the 'this.' prefix to call the method in this class?

Reply:
Good point